### PR TITLE
properly deal with submissions to cases that should be purged but haven't yet

### PR DESCRIPTION
**Pull request by czue**
context:

Cory Zue [2:26 PM] 
sorry i feel like i've asked this before but can someone clarify when case purging happens?

Cory Zue [2:26 PM]
it's not at the end of every form but before sync, yeah?

Cory Zue [2:28 PM]
context: https://www.commcarehq.org/a/spf-defaulters/reports/case_data/7a0bef45-7de7-4a01-bf08-758e6e584fc2/#!history

this case seems like it should have been immediately purged since the owner id is set to blank in the creating form, but an update form is immediately submitted right after

Cory Zue [2:28 PM]
just wanted to make sure that's an expected scenario

Cory Zue [2:34 PM] 
@ctsims: ^

Clayton Sims [2:35 PM] 
@czue: That sounds correct (re: purge happening after we sync)

Clayton Sims [2:35 PM]
er

Clayton Sims [2:35 PM]
before we sync

Clayton Sims [2:35 PM]
but not after form entry

wait for tests. @snopoke cc @proteusvacuum since you might have conflicting changes with this.

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/mock.py [hunk 1]
```diff
@@ -122,9 +122,10 @@ def __init__(self,
                 'xmlns':                self.XMLNS,
             }
         })
-        create_or_update.update({
-            'owner_id':                 owner_id,
-        })
+        if owner_id is not None:
+            create_or_update.update({
+                'owner_id':                 owner_id,
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/mock.py]: E241 multiple spaces after ':'

Comment (czue) [inline corehq/ex-submodules/casexml/apps/case/mock.py]: this whole file has super weird indentation

**Comment by snopoke** [comment]
:+1:

**Comment by czue** [comment]
(looking at test fail)
```
======================================================================
FAIL: testMultipleCases (casexml.apps.phone.tests.test_sync_mode.SyncTokenUpdateTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 45, in inner
    return helper(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 32, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/tzmigration/test_utils.py", line 28, in call_with_settings
    fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/restore_test_utils.py", line 28, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 204, in testMultipleCases
    self._testUpdate(sync_log.get_id, {"asdf": []})
  File "corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py", line 100, in _testUpdate
    self.assertEqual(set(all_ids), sync_log.case_ids_on_phone)
AssertionError: set(['asdf']) != JsonSet([])
----------------------------------------------------------------------
```

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**